fms2_io/include/domain_read.inc

```diff
@@ -86,48 +86,52 @@ end subroutine domain_read_1d
 !!        decomposed".
 subroutine domain_read_2d(fileobj, variable_name, vdata, unlim_dim_level, &
                           corner, edge_lengths)
-
-  type(FmsNetcdfDomainFile_t), intent(in) :: fileobj !< File object.
-  character(len=*), intent(in) :: variable_name !< Variable name.
-  class(*), dimension(:,:), intent(inout) :: vdata !< Data that will
-                                                   !! be written out
-                                                   !! to the netcdf file.
-  integer, intent(in), optional :: unlim_dim_level !< Level for the unlimited
-                                                   !! dimension.
-  integer, dimension(2), intent(in), optional :: corner !< Array of starting
-                                                        !! indices describing
-                                                        !! where the data
-                                                        !! will be written to.
-  integer, dimension(2), intent(in), optional :: edge_lengths !< The number of
-                                                              !! elements that
-                                                              !! will be written
-                                                              !! in each dimension.
-
-  integer :: xdim_index
-  integer :: ydim_index
-  type(domain2d), pointer :: io_domain
-  integer :: xpos
-  integer :: ypos
-  integer :: i
-  integer :: isd
-  integer :: isc
-  integer :: xc_size
-  integer :: jsd
-  integer :: jsc
-  integer :: yc_size
-  integer, dimension(:), allocatable :: pe_isc
-  integer, dimension(:), allocatable :: pe_icsize
-  integer, dimension(:), allocatable :: pe_jsc
-  integer, dimension(:), allocatable :: pe_jcsize
-  integer, dimension(2) :: c
-  integer, dimension(2) :: e
-  integer(kind=i4_kind), dimension(:,:), allocatable :: buf_i4_kind
-  integer(kind=i8_kind), dimension(:,:), allocatable :: buf_i8_kind
-  real(kind=r4_kind), dimension(:,:), allocatable :: buf_r4_kind
-  real(kind=r8_kind), dimension(:,:), allocatable :: buf_r8_kind
-  logical :: buffer_includes_halos
-  integer :: xgmin !< Starting x index of global io domain
-  integer :: ygmin !< Starting y index of global io domain
+  type(FmsNetcdfDomainFile_t),  intent(in)           :: fileobj         !< File object.
+  character(len=*),             intent(in)           :: variable_name   !< Variable name.
+  class(*), contiguous, target, intent(inout)        :: vdata(:,:)      !< Data that will
```

Comment:
I meant to mention your comments mention _writes_ instead of _reads_ here and in the 3d version.